pkg/collector/python/helpers.go

```diff
@@ -83,13 +83,18 @@ var (
 // newStickyLock registers the current thread with the interpreter and locks
 // the GIL. It also sticks the goroutine to the current thread so that a
 // subsequent call to `Unlock` will unregister the very same thread.
-func newStickyLock() *stickyLock {
+func newStickyLock() (*stickyLock, error) {
 	runtime.LockOSThread()
+
+	if rtloader == nil {
```

Comment:
Can `Destroy()` be called by another goroutine when `newStickyLock()` is executed?
In that case,  if `Destroy()` is called after line 89 but before line 93, `rtloader` may be `nil` (or the memory is freed) when calling `C.ensure_gil(rtloader)`

Reply:
👍 Yup, you're right - it's a very small edge case but it's definitely a valid one that I missed! I just pushed a fix where I added locks around the relevant sections (so it will take some time to build). I do think we may need to spend some more time in the future around rtloader lifecycle logic since not a lot of the interaction code seems like it can handle the rtloader pointer manipulation gracefully, especially in multithreaded interactions.